File: backend/api/receipt.go
package api

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Receipt struct {
	Id          primitive.ObjectID `bson:"_id" json:"id"`
	Date        time.Time          `bson:"date" json:"date"`
	Description string             `bson:"description" json:"description"`
	Entries     []ReceiptEntry     `bson:"entries" json:"entries"`
	RetailerId  string             `bson:"retailerId" json:"retailerId"`
	BranchId    string             `bson:"branchId" json:"branchId"`
	Metadata    ReceiptMetadata    `bson:"metadata" json:"metadata"`
}

File: App/app/src/main/java/com/example/balticbytebuccaneers/service/receipt/Receipt.kt
package com.example.balticbytebuccaneers.service.receipt

data class Receipt(
    val branchId: String?,
    val date: String?,
    val description: String?,
    val entries: List<Entry>?,
    val id: String?,
    val metaData: ReceiptMetaData?,
    val retailerId: String?
)
